clients/vault/src/oracle/agent.rs

```diff
@@ -57,84 +87,104 @@ async fn handle_message(
 	Ok(())
 }
 
-/// Start the connection to the Stellar Node.
-/// Returns an `OracleAgent` that will handle incoming messages from Stellar Node,
-/// and to send messages to Stellar Node
-pub async fn start_oracle_agent(
+pub async fn listen_for_stellar_messages(
 	config: StellarOverlayConfig,
-	secret_key: &str,
+	oracle_agent: ArcRwLock<OracleAgent>,
+	secret_key_as_string: String,
 	shutdown_sender: ShutdownSender,
-) -> Result<OracleAgent, Error> {
-	let is_public_network = config.is_public_network();
+) -> Result<(), service::Error<crate::Error>> {
+	tracing::info!(
+		"listen_for_stellar_messages(): Starting connection to Stellar overlay network..."
+	);
 
-	tracing::info!("start_oracle_agent(): Starting connection to Stellar overlay network...");
+	let mut overlay_conn = connect_to_stellar_overlay_network(config.clone(), secret_key_as_string)
+		.await.map_err(|e|{
+			tracing::error!("listen_for_stellar_messages(): Failed to connect to Stellar overlay network: {e:?}");
+			service::Error::StartOracleAgentError
+		})?;
 
-	let mut overlay_conn = connect_to_stellar_overlay_network(config.clone(), secret_key).await?;
 	// use StellarOverlayConnection's sender to send message to Stellar
 	let sender = overlay_conn.sender();
+	{
+		oracle_agent.write().await.message_sender = Some(sender.clone());
+	};
 
-	let collector = Arc::new(RwLock::new(ScpMessageCollector::new(
-		config.is_public_network(),
-		config.stellar_history_archive_urls(),
-	)));
-	let collector_clone = collector.clone();
-
-	let shutdown_sender_clone = shutdown_sender.clone();
-	// disconnect signal sender tells the StellarOverlayConnection to close its TcpStream to Stellar
-	// Node
-	let (disconnect_signal_sender, mut disconnect_signal_receiver) = mpsc::channel::<()>(2);
-
-	tokio::spawn(async move {
-		let sender_clone = overlay_conn.sender();
-		loop {
-			match disconnect_signal_receiver.try_recv() {
-				// if a disconnect signal was sent, disconnect from Stellar.
-				Ok(_) | Err(TryRecvError::Disconnected) => {
-					tracing::info!("start_oracle_agent(): disconnect overlay...");
-					break;
-				},
-				Err(TryRecvError::Empty) => {},
-			}
+	// log a new message received, every 1 minute.
+	let interval = Duration::from_secs(60);
+	let mut next_time = Instant::now() + interval;
+	let mut is_proof_building_ready: Option<bool> = Some(false);
+	loop {
+		let collector = oracle_agent.read().await.collector.clone();
+
+		match overlay_conn.listen().await {
+			Ok(None) => {},
+			Ok(Some(StellarMessage::Hello(_))) => {
+				tracing::info!(
+					"listen_for_stellar_messages(): received hello message from Stellar"
+				);
+				is_proof_building_ready = Some(true);
```

Comment:
If _hello_ message is received, it signals the agent that the overlay is running. 
However, it might not be ready for proof_building just yet, hence why it is a `Some()` (or why I chose `Option`  as a representation of this).

Reply:
So this is for the case when we receive a stellar message (`Some(msg)`) before to the `Hello` variant, then `handle_message` will not `trace`. Why do we need to receive first the `Hello` message to consider the proof building ready?

Reply:
This was an attempt to cleanly understand the logs. 

Looking at your pov, the hello does look like an overkill, for proof building ready. 
I will edit this portion and see what I can do. 

Thank you Gianni!

Reply:
I'm not sure I understand why we need this variable? Also, with Option<bool> it can now be three things, `None`, `Some(true)` and `Some(false)`. As far as I can tell, we only check for `Some(true)` and never for `None` so it should be fine if we don't make it an `Option` type but a simple boolean as we apparently only need the two cases.

Reply:
Okay! Yeah I also thought that a boolean works as well since we use two states but if it's for logging only, of course leave it as long as you need.

Reply:
`is_proof_building_ready` completed **_removed_**.